mintpy/createmask_cp.py

```diff
@@ -0,0 +1,142 @@
+        cp0_w = cp0_w - (phase[idx,:,:]-inps.ref_phase[idx])
+        cum_cp = cum_cp + (np.exp(1j*cp0_w))
+        
+    # cum_cp = np.angle(cum_cp)
+    return cum_cp, num_cp
+
+def main(iargs = None):
+    inps = cmd_line_parse(iargs)
+    stack_obj = ifgramStack(inps.ifgram_stack)
+    stack_obj.open()
+    length, width = stack_obj.length, stack_obj.width
+    date12_list = stack_obj.get_date12_list(dropIfgram=True)
+    date12_list_all = stack_obj.get_date12_list(dropIfgram=False)
+    print('scene length, width', length, width)
+    ref_phase = stack_obj.get_reference_phase(unwDatasetName = 'unwrapPhase')
+    inps.ref_phase = ref_phase
+    inps.length = length
+    inps.width = width
+    # retrieve the list of SLC dates from ifgramStack.h5
+    ifgram0 = date12_list[0]
+    date1, date2 = ifgram0.split('_')
+    SLC_list = [date1, date2]
+    for ifgram in date12_list:
+        date1, date2 = ifgram.split('_')
+        if date1 not in SLC_list:
+            SLC_list.append(date1)
+        if date2 not in SLC_list:
+            SLC_list.append(date2)
+    SLC_list.sort()
+    print('number of SLC found : ', len(SLC_list))
+    print('first SLC: ', SLC_list[0])
+    print('last  SLC: ', SLC_list[-1])
+    inps.SLC_list = SLC_list
+    inps.NSLC = len(SLC_list)
+    inps.date12_list = date12_list
+    inps.date12_list_all = date12_list_all
+
+    # split igram_file into blocks to save memory
+    box_list, num_box = ifginv.split2boxes(inps.ifgram_stack,inps.max_memory)
+    closurephase =  np.zeros([length,width],np.complex64)
+    #process block-by-block
+    for i, box in enumerate(box_list):
+            box_width  = box[2] - box[0]
+            box_length = box[3] - box[1]
+            print(box)
+            if num_box > 1:
+                print('\n------- processing patch {} out of {} --------------'.format(i+1, num_box))
+                print('box width:  {}'.format(box_width))
+                print('box length: {}'.format(box_length))
+
+            closurephase[box[1]:box[3],box[0]:box[2]], numcp = cum_seq_closurePhase(inps,inps.nl,box)
+
+    
+    
+    # What is a good threadshod?
+    # Assume that it's pure noise so that the phase is uniform distributed from -pi to pi.
+    # The standard deviation of phase in each loop is pi/sqrt(3) (technically should be smaller because when forming loops there should be a reduction in phase variance)
+    # The standard deviation of phase in cumulative wrapped closure phase is pi/sqrt(3)/sqrt(numcp) -- again another simplification assuming no correlation.
+    # We use 2\delta as threashold -- 95.4% confidence
+
+    threashold_cp = np.pi/np.sqrt(3)/np.sqrt(numcp)*3 # 3/sigma, 99.7% confidence
+    mask = np.ones([length,width],np.float32)
+    mask[np.abs(np.angle(closurephase))>threashold_cp] = 0 # this masks areas with potential bias
+    mask[np.abs(np.abs(closurephase)/numcp < 0.3)] = 1 # this unmasks areas with low correlation (where it's hard to know wheter there is bias either)
```

Comment:
How about exposing the magic threshold of 0.3 to the users and default it to 0.3 instead of hardcoding it.